packages/react-dom/src/__tests__/ReactErrorBoundaries-test.internal.js

```diff
@@ -1039,29 +1225,42 @@ describe('ReactErrorBoundaries', () => {
       'ErrorBoundary static getDerivedStateFromError',
       'ErrorBoundary componentWillMount',
       'ErrorBoundary render error',
+      'ErrorBoundary constructor',
```

Comment:
I added these expected log statements to make the tests pass I know, `createRoot(..).render` is more efficient than `ReactDOM.render` so I assumed this is a byproduct of these functional differences but that could be wrong.

Reply:
Yeah this is kind of expected, since in `createRoot` we retry rendering when we catch an error. But I didn't know we did this in error boundaries, I can't think of a reason we want to do this when rendering error boundaries, since the content in boundaries should be static? cc @acdlite

Also can you add some comments that indicate that these logs are for the error retry like we have elsewhere?

Reply:
Talked to @acdlite, this is intentional, we retry in both error boundaries and suspense fallbacks because you may depend on some concurrent state like from a store in either case. The reason we do retries is to catch errors with things like tearing which will succeed in the next render, and that reasoning applies to fallbacks and boundaries too.

Reply:
Added comments indicating that these logs are for error retry everywhere that its relevant in this file.